### PR TITLE
Data erasure support

### LINKED ISSUE #1757: Data Erasure of Personal Data support
Adding support for personal data delete.

### PR DESCRIPTION
**Pull request by acbeni**
# Description

Closes #1757

# Checklist

- [x] Update release Notes
- [x] Add to the current github milestone

### CODE REVIEW

File: commercetools-models/src/main/java/io/sphere/sdk/customobjects/commands/CustomObjectDeleteCommandImpl.java [hunk 1]
```diff
@@ -30,29 +30,49 @@ public HttpRequestIntent httpRequestIntent() {
...
+    CustomObjectDeleteCommandImpl(final String id, final Long version,final boolean eraseData , final JavaType valueJavaType) {
+        this.path = format("/%s?version=%d%s", id, version,eraseData? "&dataErasure=true":"");
+        this.javaType = CustomObjectUtils.getCustomObjectJavaTypeForValue(valueJavaType);
     }
 
-    CustomObjectDeleteCommandImpl(final String container, final String key, final JavaType valueJavaType) {
-        this.path = format("/%s/%s", container, key);
-        this.javaType = CustomObjectUtils.getCustomObjectJavaTypeForValue(valueJavaType);
+    CustomObjectDeleteCommandImpl(final String container, final String key, final Class<T> valueClass) {
+        this(container, key,true, SphereJsonUtils.convertToJavaType(valueClass));
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/customobjects/commands/CustomObjectDeleteCommandImpl.java]: According to our dev documentation, the default for `eraseData` is `false`. And I would expect that our JVM SDK uses this default here too.

File: commercetools-models/src/main/java/io/sphere/sdk/customobjects/commands/CustomObjectDeleteCommandImpl.java [hunk 2]
```diff
@@ -30,29 +30,49 @@ public HttpRequestIntent httpRequestIntent() {
...
+    }
+
+    CustomObjectDeleteCommandImpl(final String container, final String key, final TypeReference<T> typeReference) {
+        this(container, key,true, SphereJsonUtils.convertToJavaType(typeReference));
     }
 
     CustomObjectDeleteCommandImpl(final String id, final Long version, final Class<T> valueClass) {
-        this(id, version, SphereJsonUtils.convertToJavaType(valueClass));
+        this(id, version,true, SphereJsonUtils.convertToJavaType(valueClass));
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/customobjects/commands/CustomObjectDeleteCommandImpl.java]: Same as above.

File: commercetools-models/src/main/java/io/sphere/sdk/customobjects/commands/CustomObjectDeleteCommandImpl.java [hunk 3]
```diff
@@ -30,29 +30,49 @@ public HttpRequestIntent httpRequestIntent() {
...
     }
 
     CustomObjectDeleteCommandImpl(final String id, final Long version, final TypeReference<T> typeReference) {
-        this(id, version, SphereJsonUtils.convertToJavaType(typeReference));
+        this(id, version, true,SphereJsonUtils.convertToJavaType(typeReference));
```
Comment (katmatt) [inline commercetools-models/src/main/java/io/sphere/sdk/customobjects/commands/CustomObjectDeleteCommandImpl.java]: Same as above.

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/commands/MetaModelByIdDeleteCommandImpl.java [hunk 4]
```diff
@@ -28,27 +28,33 @@
...
         this.expansionPaths = requireNonNull(expansionPaths);
         this.versioned = requireNonNull(versioned);
         this.endpoint = requireNonNull(endpoint);
         this.javaType = requireNonNull(javaType);
+        this.eraseData = eraseData;
     }
 
     protected MetaModelByIdDeleteCommandImpl(final Versioned<T> versioned, final JsonEndpoint<T> endpoint, final E expansionModel, final Function<MetaModelByIdDeleteCommandBuilder<T, C, E>, C> creationFunction) {
-        this(versioned, convertToJavaType(endpoint.typeReference()), endpoint.endpoint(), expansionModel, Collections.emptyList(), creationFunction);
+        this(versioned, false, convertToJavaType(endpoint.typeReference()), endpoint.endpoint(), expansionModel, Collections.emptyList(), creationFunction);
```
Comment (katmatt) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/commands/MetaModelByIdDeleteCommandImpl.java]: And here we are using `false`, which is the value that I expected 👍

File: commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasDeleteCommand.java [hunk 5]
```diff
@@ -10,4 +10,7 @@
     String[] includeExamples() default {};
 
     String[] deleteWith() default {};
+
+    boolean canEraseUsersData() default false;
```
Comment (katmatt) [inline commercetools-sdk-base/src/main/java/io/sphere/sdk/annotations/HasDeleteCommand.java]: Nice 👍

**Comment by acbeni** [comment]
Thanks @katmatt for the review, i now submitted the requested changes 👍

**Comment by katmatt** [comment]
Looks very good to me 👍

**Pull request merged**